gcc/rust/util/rust-unicode.cc

```diff
@@ -39,75 +39,35 @@ const uint32_t L_COUNT = 19, V_COUNT = 21, T_COUNT = 28;
 const uint32_t N_COUNT = V_COUNT * T_COUNT;
 const uint32_t S_COUNT = L_COUNT * N_COUNT;
 
+// Check if the codepoint is in any of the ranges (half-open intervals [a,b)).
 template <std::size_t SIZE>
-int64_t
+bool
 binary_search_ranges (
-  // FIXME: use binray search function from <algorithm>
   const std::array<std::pair<uint32_t, uint32_t>, SIZE> &ranges,
   uint32_t target_cp)
 {
-  if (SIZE == 0)
-    return -1;
-
-  uint32_t low, high, mid;
-  uint32_t start, end;
-  low = 0;
-  high = SIZE - 1;
-  mid = (low + high) / 2;
-  while (high - low > 1)
-    {
-      start = ranges[mid].first;
-      end = ranges[mid].second;
-      if (start <= target_cp && target_cp < end)
-	{
-	  return mid;
-	}
-      else if (end <= target_cp)
-	low = mid + 1;
-      else
-	high = mid - 1;
-      mid = (low + high) / 2;
-    }
-
-  if (ranges[mid].first <= target_cp && target_cp < ranges[mid].second)
-    return mid;
+  auto it = std::lower_bound (ranges.begin (), ranges.end (), target_cp,
+			      [] (const std::pair<uint32_t, uint32_t> &a,
+				  uint32_t b) { return a.second <= b; });
+  if (it == ranges.end ())
+    return false;
   else
-    return -1;
+    return it->first <= target_cp && target_cp < it->second;
 }
 
 template <std::size_t SIZE>
-int64_t
+bool
```

Comment:
Should we replace this function with a direct call to std::binary_search instead ?

Reply:
Oh yes. i
I will fix it later